### ISSUE TITLE
scriptWithInlineFile() doesnt work for reading resources within jar rather than actual files

### ISSUE THREAD

**Issue by ijabz** [tipsy/j2html#108]
This causes issue referred to by :

https://github.com/perwendel/spark/issues/917
https://stackoverflow.com/questions/48443719/best-way-to-create-readable-javascript-in-java-for-adding-to-html-page

Looking at InlineStaticResource you can see that even if we retrieve the file as a resource (i.e from a jar) we then try to load it as a file in the _readFileAsString_ method, but its not a file its a stream !

```
public static String getFileAsString(String path) {
       try {
           return readFileAsString(InlineStaticResource.class.getResource(path).getPath());
       } catch (Exception var4) {
           try {
               return readFileAsString(path);
           } catch (Exception var3) {
               throw new RuntimeException("Couldn't find file with path='" + path + "'");
           }
       }
   }

   private static String readFileAsString(String path) throws IOException {
       BufferedReader bufferedReader = new BufferedReader(new FileReader(path));
       StringBuilder sb = new StringBuilder();

       int c;
       while((c = bufferedReader.read()) >= 0 && c >= 0) {
           sb.append((char)c);
       }

       return sb.toString();
   }
```

I added the following to my code to demonstrate this
```
try
{
System.out.println(""+InlineStaticResource.class.getResource("/artwork.js").toString());
System.out.println(""+InlineStaticResource.class.getResource("/artwork.js").getPath());
System.out.println("*************"+new FileReader(InlineStaticResource.class.getResource("/artwork.js").getPath()));
}
catch(Exception e)
{
MainWindow.logger.log(Level.SEVERE, e.getMessage(), e);
}

```
Giving output as follows

```
*************jar:file:/C:/Program%20Files/Jthink/SongKong/lib/songkong-5.2.jar!/artwork.js
*************file:/C:/Program%20Files/Jthink/SongKong/lib/songkong-5.2.jar!/artwork.js
Mar 14, 2018 3:00:39 PM com.jthink.songkong.cmdline.SongKong main
SEVERE: file:\C:\Program%20Files\Jthink\SongKong\lib\songkong-5.2.jar!\artwork.js (The filename, directory name, or volume label syntax is incorrect)
java.io.FileNotFoundException: file:\C:\Program%20Files\Jthink\SongKong\lib\songkong-5.2.jar!\artwork.js (The filename, directory name, or volume label syntax is incorrect)
at java.io.FileInputStream.open0(Native Method)
at java.io.FileInputStream.open(Unknown Source)
at java.io.FileInputStream.(Unknown Source)
at java.io.FileInputStream.(Unknown Source)
at java.io.FileReader.(Unknown Source)
at com.jthink.songkong.cmdline.SongKong.main(SongKong.java:2412)

```
As you can see it finds the resource in the jar okay, the problem is when it trys to read it as a file, instead I think it should use InputStreamReader

**Comment by tipsy** [comment]
Hi @ijabz, is this on `1.2.2`? I thought this issue was fixed already.

**Comment by tipsy** [comment]
Should have been fixed in #84

**Comment by ijabz** [comment]
Ah, you are quite right I was on 1.2.0 now working, (confused myself because originally thought the problem was with spark framework and I already had latest version of that so didnt think to look at j2html versions)
